app/lib/main.dart

```diff
@@ -34,15 +35,31 @@ Future<void> startAppInner() async {
     final license = await rootBundle.loadString('google_fonts/LICENSE.txt');
     yield LicenseEntryWithLineBreaks(['google_fonts'], license);
   });
+  await initializeNotifications();
   runApp(const ProviderScope(child: Acter()));
 }
 
-class Acter extends ConsumerWidget {
-  const Acter({Key? key}) : super(key: key);
+class Acter extends ConsumerStatefulWidget {
+  static const String name = 'Awesome Notifications - Example App';
+  static const Color mainColor = Colors.deepPurple;
+
+  const Acter({super.key});
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => _ActerState();
+}
+
+class _ActerState extends ConsumerState<Acter> {
+  @override
+  void initState() {
+    setupNotificationsListeners();
```

Comment:
The most recommended way is to have this line below ```super.initState()``` as we should have everything else after framework builds widget and vice versa in case of ```super.dispose()```